examples/step-4/doc/results.dox

```diff
@@ -100,11 +100,324 @@ wants to see, but no more.
+the solution varies between values of 1 and 2, but with a larger part of the
+solution closer to one than two; so an average value of 1.33 for the mean value
+is reasonable. For the flux, recall that $\nabla u \cdot \mathbf n$ is the
+directional derivative in the normal direction -- in other words, how the
+solution changes as we move from the interior of the domain towards the
+boundary. If you look at the 2d solution, you will realize that for most parts
+of the boundary, the solution *decreases* as we approach the boundary, so the
+normal derivative is negative -- so if we integrate along the boundary, we
+should expect (and obtain!) a negative value.
+
+
+
 <a name="extensions"></a>
 <h3>Possibilities for extensions</h3>
 
+There are many ways with which one can play with this program. The simpler
+ones include essentially all the possibilities already discussed in the
+<a href="step_3.html#extensions" target="body">Possibilities for extensions in the documentation of step 3</a>,
+except that you will have to think about whether something now also applies
+to the 3d case discussed in the current program.
+
+It is also worthwhile considering the postprocessing options discussed
+above. The documentation states two numbers (the mean value and the
+normal flux) for both the 2d and 3d cases. Can we trust these
+numbers? We have convinced ourselves that at least the mean value
+is reasonable, and that the sign of the flux is probably correct.
+But are these numbers accurate?
+
+A general rule is that we should never trust a number unless we have
+verified it in some way. From the theory of finite element methods,
+we know that as we make the mesh finer and finer, the numerical
+solution $u_h$ we compute here must converge to the exact solution
+$u$. As a consequence, we also expect that $\bar u_h \rightarrow \bar u$
+and $\Phi_h \rightarrow \Phi$, but that does not mean that for any
+given mesh $\bar u_h$ or $\Phi_h$ are particularly accurate approximations.
+
+To test this kind of thing, we have already considered the convergence of
+a point value in step-3. We can do the same here by selecting how many
+times the mesh is globally refined in the `make_grid()` function of this
+program. For the mean value of the solution, we then get the following
+numbers:
+  <table align="center" class="doxtable">
+    <tr> <th># of refinements</th>
+         <th>$\bar u_h$ in 2d</th>
+         <th>$\bar u_h$ in 3d</th>
+    </tr>
+    <tr> <td>4</td> <td>1.33303</td> <td>1.58058</td> </tr>
+    <tr> <td>5</td> <td>1.33276</td> <td>1.57947</td> </tr>
+    <tr> <td>6</td> <td>1.3327</td>  <td>1.5792</td> </tr>
+    <tr> <td>7</td> <td>1.33269</td> <td>1.57914</td> </tr>
+    <tr> <td>8</td> <td>1.33268</td> <td></td> </tr>
+    <tr> <td>9</td> <td>1.33268</td> <td></td> </tr>
+  </table>
+I did not have the patience to run the last two values for the 3d case --
+one needs quite a fine mesh for this, with correspondingly long run times.
```

Comment:
I think this statement is repeated further down (where I suggest to delete it altogether), and I do not like this. I also think that the problem with 16 million unknowns will hardly fit in memory, whereas the next one with 135m DoFs will certainly not in a typical laptop as it would take around 80GB for just the matrix. I suggest the following:
```suggestion
We did not run the solver for the last two values for the 3d case,
because as the mesh is refined the number of unknowns grows quickly.
Then, the run times would be unacceptably high with the basic iterative
solver chosen here. In fact, it is not optimal for Poisson-type problems;
for better solvers, see, e.g., step-16 or step-37.
```

Reply:
I have wanted to add something about solvers elsewhere (https://dealii.org/developer/doxygen/deal.II/step_6.html#Solversandpreconditioners is the right place). So let me skip this suggestion here and instead pick it up in step-6.